onmt/translate/Translator.lua

```diff
@@ -167,84 +169,87 @@ function Translator:translateBatch(batch)
   self.models.decoder:maskPadding()
 
   local encStates, context = self.models.encoder:forward(batch)
-
-  -- Compute gold score.
-  local goldScore
-  if batch.targetInput ~= nil then
-    if batch.uneven then
-      self.models.decoder:maskPadding(batch.sourceSize, batch.sourceLength)
+  if self.opt.print_input_embeddings then
+    return encStates
```

Comment:
`encStates` contains the final cell and hidden states of all layers. Don't you want to just return the last hidden state `encStates[#encStates]`?